Stripe/STPPostalCodeValidator.m

```diff
@@ -7,40 +7,171 @@
 //
 
 #import "STPPostalCodeValidator.h"
+
 #import "STPCardValidator.h"
 #import "STPPhoneNumberValidator.h"
+#import "NSCharacterSet+Stripe.h"
+#import "NSCharacterSet+Stripe.h"
+#import "NSString+Stripe.h"
+
+static NSString *const STPCountryCodeUnitedStates = @"US";
 
 @implementation STPPostalCodeValidator
 
-+ (BOOL)stringIsValidPostalCode:(nullable NSString *)string
-                           type:(STPPostalCodeType)postalCodeType {
-    switch (postalCodeType) {
-        case STPCountryPostalCodeTypeNumericOnly: {
-            NSInteger length = [STPCardValidator sanitizedNumericStringForString:string].length;
-            return (length == 5 || length == 9);
++ (STPCardValidationState)validationStateForPostalCode:(NSString *)postalCode
+                                           countryCode:(NSString *)countryCode {
+    NSString *sanitizedCountryCode = countryCode.uppercaseString;
+    if ([self postalCodeIsRequiredForCountryCode:countryCode]) {
+        if ([sanitizedCountryCode isEqualToString:STPCountryCodeUnitedStates]) {
+            return [self validationStateForUSPostalCode:postalCode];
+        }
+        else {
+            if (postalCode.length > 0) {
+                return STPCardValidationStateValid;
+            }
+            else {
+                return STPCardValidationStateIncomplete;
+            }
         }
-        case STPCountryPostalCodeTypeAlphanumeric:
-            return string.length > 0;
-        case STPCountryPostalCodeTypeNotRequired:
-            return YES;
+    }
+    else {
+        return STPCardValidationStateValid;
```

Comment:
I think `validationStateForPostalCode: countryCode:` and `validationStateForUSPostalCode:` readability could benefit from some early returns but this is totally valid code too.